src/pf/anf.lua

```diff
@@ -247,7 +247,8 @@ local function renumber(expr)
    end
    local function lookup(var)
       if var == 'len' then return var end
-      return assert(substs[var], "unbound variable: "..var)
+      local subst = assert(substs[var], "unbound variable: "..var)
+      return subst
```

Comment:
I see, and if if the last element in a table initializer is a visit(), that splats the multiple return values into the table, is that right?  In that case LGTM and what a mess!